src/components/scenes/FioDomainRegisterSelectWalletScene.js

```diff
@@ -0,0 +1,296 @@
+    supportedCurrencies: {},
+    paymentInfo: {}
+  }
+
+  componentDidMount(): void {
+    this.getRegInfo()
+  }
+
+  getRegInfo = async () => {
+    this.setState({ loading: true })
+
+    try {
+      const { activationCost, supportedCurrencies, paymentInfo } = await getDomainRegInfo(
+        this.props.fioPlugin,
+        this.props.fioDomain,
+        this.props.selectedWallet,
+        this.props.fioDisplayDenomination
+      )
+      this.setState({ activationCost, supportedCurrencies, paymentInfo })
+    } catch (e) {
+      showError(e.message)
+    }
+
+    this.setState({ loading: false })
+  }
+
+  onPressNext = async () => {
+    const { supportedCurrencies } = this.state
+
+    const allowedCurrencyCodes = []
+    for (const currency in supportedCurrencies) {
+      if (supportedCurrencies[currency]) {
+        allowedCurrencyCodes.push(currency)
+      }
+    }
+    Airship.show(bridge => <WalletListModal bridge={bridge} headerTitle={s.strings.select_wallet} allowedCurrencyCodes={allowedCurrencyCodes} />).then(
+      ({ walletId, currencyCode }: WalletListResult) => {
+        if (walletId && currencyCode) {
+          this.onSelectWallet(walletId, currencyCode)
+        }
+      }
+    )
+  }
+
+  onSelectWallet = async (walletId: string, paymentCurrencyCode: string) => {
+    const { isConnected, selectedWallet, fioDomain, state } = this.props
+    const { activationCost, paymentInfo: allPaymentInfo } = this.state
+
+    if (isConnected) {
+      if (paymentCurrencyCode === Constants.FIO_STR) {
+        const { fioWallets } = this.props
+        const paymentWallet = fioWallets.find(fioWallet => fioWallet.id === walletId)
+        Actions[Constants.FIO_DOMAIN_CONFIRM]({
+          fioName: fioDomain,
+          paymentWallet,
+          fee: activationCost,
+          ownerPublicKey: selectedWallet.publicWalletInfo.keys.publicKey
+        })
+      } else {
+        this.props.onSelectWallet(walletId, paymentCurrencyCode)
+
+        const exchangeDenomination = getExchangeDenomination(state, paymentCurrencyCode)
+        let nativeAmount = bns.mul(allPaymentInfo[paymentCurrencyCode].amount, exchangeDenomination.multiplier)
+        nativeAmount = bns.toFixed(nativeAmount, 0, 0)
+
+        const guiMakeSpendInfo = {
+          currencyCode: paymentCurrencyCode,
+          nativeAmount,
+          publicAddress: allPaymentInfo[paymentCurrencyCode].address,
+          metadata: {
+            name: s.strings.fio_address_register_metadata_name,
+            notes: `${s.strings.title_register_fio_domain}\n${fioDomain}`
+          },
+          dismissAlert: true,
+          lockInputs: true,
+          onDone: (error: Error | null, edgeTransaction?: EdgeTransaction) => {
+            if (error) {
+              setTimeout(() => {
+                showError(s.strings.create_wallet_account_error_sending_transaction)
+              }, 750)
+            } else if (edgeTransaction) {
+              Alert.alert(
```

Comment:
Do not use the system alert. Use an airship modal instead. ButtonModal is ideal and easy to use.